src/main/scala/viper/gobra/frontend/Parser.scala

```diff
@@ -1223,15 +1223,25 @@ object Parser {
       }
 
     lazy val implementationProof: Parser[PImplementationProof] =
-      (typ <~ "implements") ~ typ ~ ("{" ~> (methodImplementationProof <~ eos).* <~ "}").? ^^ {
-        case subT ~ superT ~ memberProofOpt => PImplementationProof(subT, superT, memberProofOpt.getOrElse(Vector.empty))
+      (typ <~ "implements") ~ typ ~ ("{" ~> (implementationProofPredicateAlias <~ eos).* ~ (methodImplementationProof <~ eos).* <~ "}").? ^^ {
+        case subT ~ superT ~ Some(predAlias ~ memberProof) =>PImplementationProof(subT, superT, predAlias, memberProof)
+        case subT ~ superT ~ None => PImplementationProof(subT, superT, Vector.empty, Vector.empty)
       }
 
+    lazy val implementationProofPredicateAlias: Parser[PImplementationProofPredicateAlias] =
+      ("pred" ~> idnUse <~ ":=") ~ (selection | namedOperand) ^^ PImplementationProofPredicateAlias
```

Comment:
this syntax dissonates a bit from the rest of the language because (1) the type annotation occurs before the identifier (2) allows a type annotation in the `:=` syntax, which is not supported otherwise. I do NOT oppose this syntax strongly because this is a very particular case. However, I would suggest instead allowing the short var declaration syntax without the type annotation `perm` (wouldn't we be able to parse this case unambiguously given that every assignment in an implementation proof must be to a perm alias?)

Reply:
PS: `pred` is not exactly a type annotation, but I hope that my point is still valid